scripts/provision.sh

```diff
@@ -200,7 +201,41 @@ start_datastore() {
 }
 
 start_minio() {
-  sudo hab svc load "${BLDR_ORIGIN}/builder-minio" --channel "${BLDR_MINIO_CHANNEL:=$BLDR_CHANNEL}" --force
+  
+  set +e
+  is_minio_migration_needed
+  migration_needed=$?
+  set -e
+  
+  if [ "$migration_needed" -eq 1 ]; then
+    sudo hab svc load "${BLDR_ORIGIN}/builder-minio" --channel "stable" --force
+    bash ./minio-update.sh preflight_checks
+    echo MinIO migration required
+    backup_minio_data
+    echo starting minio
+
+    bash ./minio-update.sh download
+    sudo sh -c "find /hab/svc/builder-minio/data/ -maxdepth 1 -mindepth 1 -type d | xargs rm -rf"
+    sudo hab svc unload "${BLDR_ORIGIN}/builder-minio"
+  fi
+
+  response=$(curl -s "https://bldr.habitat.sh/v1/depot/channels/$BLDR_ORIGIN/$BLDR_CHANNEL/pkgs/builder-minio/latest")
+  channel_version=$(echo "$response" | jq -r '.ident.version')
+  channel_release=$(echo "$response" | jq -r '.ident.release')
+  sudo hab svc load "${BLDR_ORIGIN}/builder-minio/$channel_version/$channel_release" --channel "${BLDR_MINIO_CHANNEL:=$BLDR_CHANNEL}" --force
+
```

Comment:
you shouldn't need to query for the latest version and release and you can get rid of `BLDR_MINIO_CHANNEL` because we don't use that anywhere. So this can be:
 sudo hab svc load "${BLDR_ORIGIN}/builder-minio" --channel "${BLDR_CHANNEL}" --force

Reply:
not specifying the /$channel_version/$channel_release loads the already existing builder-minio ie 2018

Reply:
We should delete `/hab/pkgs/habitat/builder-minio` first since we are now done with that version. That should also take care of the above.